libraries/Wire/src/Wire.cpp

```diff
@@ -501,6 +503,27 @@ inline void TwoWire::resetTxBuffer(void)
   }
 }
 
+// Send STOP sequence to recover bus.
+// Useful in case of bus stuck after a reset for example
+void TwoWire::recoverBus(void)
+{
+  pinMode(pinNametoDigitalPin(_i2c.sda), OUTPUT);
+  pinMode(pinNametoDigitalPin(_i2c.scl), OUTPUT);
+
+  for (int i = 0; i < 10; i++) {
+    digitalWrite(pinNametoDigitalPin(_i2c.sda), LOW);
+    delay(3);
+    digitalWrite(pinNametoDigitalPin(_i2c.scl), HIGH);
+    delay(3);
+    digitalWrite(pinNametoDigitalPin(_i2c.sda), HIGH);
+    delay(3);
+    digitalWrite(pinNametoDigitalPin(_i2c.scl), LOW);
```

Comment:
```suggestion
    digitalWriteFast(_i2c.sda, LOW);
    delay(3);
    digitalWriteFast(_i2c.scl, HIGH);
    delay(3);
    digitalWriteFast(_i2c.sda, HIGH);
    delay(3);
    digitalWriteFast(_i2c.scl, LOW);
```

Reply:
Yes, it it is better. Done

Reply:
@ABOSTM  maybe add also isStuck() method ?

Reply:
> 

What did you mean?
Check if SDA held low ?

Reply:
@fpistm exactly 👍

Reply:
Here the goal is to force bus recovery before Wire instance init. The main use case is when the board is reset during an I2C transfer. The method is not public.